### PR TITLE
add model cache after loaded

### PR DESCRIPTION
**Pull request by efwfe**
> Hello, I made a PR to cache model after loaded. save the model instance in memory, that's really will cost some memory here, but dont worry, it will check the memory free both cpu and gpu, and free the memory when it's necessary. 
And this is the update version of https://github.com/comfyanonymous/ComfyUI/pull/3545.

Here is a gif example to show it, and it's really amazing.

_Fully tested on openart template workflows. -> https://openart.ai/workflows/templates_

530pgpif

### CODE REVIEW

File: comfy/sd.py [hunk 1]
```diff
@@ -464,48 +465,70 @@ def load_checkpoint_guess_config(ckpt_path, output_vae=True, output_clip=True, o
...
+        model_cache.cache_model(ckpt_path, model_patcher)
+    if clip:
+        logging.debug(f"cache clip of : {ckpt_path}")
+        model_cache.cache_clip(ckpt_path, clip_key, clip)
+    if vae:
+        logging.debug(f"cache vae of : {ckpt_path}")
+        model_cache.cache_vae(ckpt_path, vae)
+    if clipvision:
+        logging.debug(f"cache clipvision of : {ckpt_path}")
+        model_cache.cache_clipvision(ckpt_path, clipvision)
```
Comment (alboto) [inline comfy/sd.py]: A bug here
`model_cache.cache_clipvision` needs to be changed to `model_cache.cache_clip_vision`

Comment (jasstionzyf) [reply comfy/sd.py]: It is better to cache clipVision model also within CLIPVisionLoader
When use ip adapter every time will need to load CLIP-ViT-H-14-laion2B-s32B-b79K.safetensors 
Use model cache can directly copy it each time from cpu memory

Comment (efwfe) [reply comfy/sd.py]: copy that.

**Comment by 333caowei** [comment]
looks great !  how to use it?

**Comment by efwfe** [comment]
> looks great ! how to use it?

You can try with this repo repository `git@github.com:efwfe/ComfyUI.git`, Thank you for wanting to use.

**Comment by efwfe** [comment]
> Hi can you keep your fork up to date please? I want to test this.

Welcome to test it. It has been updated. Hope to know your feedback. It is recommended to add `--low-vram` parameter to reduce the memory usage of sdxl models.

**Comment by shawnL128Po** [comment]
Sorry for asking a silly question. I couldn't find the definition of the variable 'ckpt_path' in the function 'load_state_dict_guess_config' within the sd.py file, and PyCharm is reporting a compilation error: 'Unresolved reference 'ckpt_path''.

**Comment by efwfe** [comment]
> Sorry for asking a silly question. I couldn't find the definition of the variable 'ckpt_path' in the function 'load_state_dict_guess_config' within the sd.py file, and PyCharm is reporting a compilation error: 'Unresolved reference 'ckpt_path''.

Sorry,that's my mistake, the latest version of comfyui remove the parameter of `ckpt_path`, I have updated the code, you can try it now.

Actually I'm ready to close this PR because it's not really helpful in most cases. So I closed this thanks you guys.

**Comment by WSvisionG** [comment]
I think it is very useful, I hope you can continue to update, thank you, cool

**Comment by chen1yunan** [comment]
Hi! Nice project and solution!

I see your workflow, do nothing but cache cpt. so if I want to use it, i just update your script into comfyui and start it, right?

Any other operations should do?

**Comment by dailingx** [comment]
hi, I think it is very useful! Thank you for your great work!
I want to know if it is still available now and if it supports caching for the LoRa model?

**Pull request closed without merging**